README.md

```diff
@@ -33,114 +34,100 @@ Built on top of the tree is the `Router` interface:
 
 ```go
 // Register a middleware handler (or few) on the middleware stack
-Use(middlewares ...interface{})
+Use(middlewares ...func(http.Handler) http.Handler
 
-// Register a new middleware stack
-Group(fn func(r Router)) Router
-
-// Mount an inline sub-router
+// Mount a sub-router along a pattern
 Route(pattern string, fn func(r Router)) Router
 
+// Register a new inline-Mux, which offers a fresh middleware stack
+Group(fn func(r Router)) Router
+
 // Mount a sub-router
-Mount(pattern string, handlers ...interface{})
+Mount(pattern string, subrouter Router)
 
 // Register routing handler for all http methods
-Handle(pattern string, handlers ...interface{})
+Handle(pattern string, handler http.Handler)
+
+// Register routing handler func for all http methods
+HandleFunc(pattern string, handler http.HandlerFunc)
 
 // Register routing handler for CONNECT http method
-Connect(pattern string, handlers ...interface{})
+Connect(pattern string, handler http.HandlerFunc)
 
 // Register routing handler for HEAD http method
-Head(pattern string, handlers ...interface{})
+Head(pattern string, handler http.HandlerFunc)
 
 // Register routing handler for GET http method
-Get(pattern string, handlers ...interface{})
+Get(pattern string, handler http.HandlerFunc)
```

Comment:
Ugh, why didn't I think of that!
![image](https://cloud.githubusercontent.com/assets/2247982/17214612/2a7c9ac8-551d-11e6-9cc5-dae493c617e6.png)